pkgs/top-level/all-packages.nix

```diff
@@ -2366,7 +2366,13 @@ with pkgs;
 
   kitty = darwin.apple_sdk_11_0.callPackage ../applications/terminal-emulators/kitty {
     harfbuzz = harfbuzz.override { withCoreText = stdenv.isDarwin; };
-    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa CoreGraphics Foundation IOKit Kernel UniformTypeIdentifiers OpenGL UserNotifications;
```

Comment:
Why were the removed frameworks removed? Are they unnecessary?

Reply:
darwin.apple_sdk_11_0.callPackage brings them into scope, no need for the inherit.

Reply:
@adamcstephens, the removed frameworks are dependencies of the remaining frameworks. (See https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/apple-sdk-11.0/frameworks.nix.)

Reply:
@NickCao, huh, I didn't know that. I'm away from my computer until tomorrow, but I'll make that change then if this PR hasn't been merged by then. (I don't think this tweak should block merging the PR though, since I think it would be good to get this merged ASAP for the sake of Kitty users on `x86_64-darwin`.)

Reply:
@NickCao, now that I'm at my machine and I've tried a few things that didn't work, I no longer think I understand what you mean. I looked at a bunch of other uses of `darwin.apple_sdk_11_0.callPackage ` in `all-packages.nix`, and they all seem to be using `inherit` like we are here. Maybe you could provide a more detailed explanation or point me to an example?

(I still don't think this should block merging this PR though.)

Reply:
I thought that `darwin.apple_sdk_11_0.callPackage ({ Libsystem }: Libsystem) { }` would work like `callPackage ({ nano }: nano) { }` do.